Expensify/App
# Upgrade react-navigation to 6.x

@marcaaron @parasharrajat 

### Details
Upgrade react-navigation to 6.x and reanimated library to 2.x to resolve issues caused by legacy drawer and animation implementations.

### Fixed Issues
Fixes https://github.com/Expensify/Expensify.cash/issues/2180 and https://github.com/Expensify/Expensify.cash/issues/2447

### Tests

Tested navigating between chats and other pages like settings, search and made sure  they work as expected.

On mobile devices changed device orientation and observed that the components are positioned properly.

### QA Steps

- Login to expensify.cash
- Navigate between chats and access all other pages (Search, settings, user profiles)
- Check if chats are visible on the background when modal pages are opened
- Verify that navigation and page animations are the same as the production

Try to reproduce these issues as instructed in the issue descriptions

- https://github.com/Expensify/Expensify.cash/issues/2447
- https://github.com/Expensify/Expensify.cash/issues/2180

### Tested On


### Screenshots

#### Web

https://user-images.githubusercontent.com/3853003/120941364-4e336400-c71a-11eb-8c47-95b984186a6d.mov

#### Mobile Web

https://user-images.githubusercontent.com/3853003/120941430-a4a0a280-c71a-11eb-885a-d0f4f1e35b01.mov

#### Desktop

https://user-images.githubusercontent.com/3853003/120941700-f72e8e80-c71b-11eb-9a63-81cdd907a02e.mp4

#### iOS

https://user-images.githubusercontent.com/3853003/119233427-ec230e00-bb20-11eb-8de9-657ccaa19ce7.mov

#### Android

https://user-images.githubusercontent.com/3853003/119234174-5e492200-bb24-11eb-9f7b-2214d27c8346.mov
